code/modules/mob/living/ventcrawling.dm

```diff
@@ -92,22 +92,22 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, typecacheof(list(
 	if(!totalMembers.len)
 		return
 
-	if(client)
-		for(var/X in totalMembers)
-			var/obj/machinery/atmospherics/A = X //all elements in totalMembers are necessarily of this type.
-			if(in_view_range(client.mob, A))
-				if(!A.pipe_vision_img)
-					A.pipe_vision_img = image(A, A.loc, dir = A.dir)
-					A.pipe_vision_img.plane = ABOVE_HUD_PLANE
-				client.images += A.pipe_vision_img
-				pipes_shown += A.pipe_vision_img
+	if(client) //MONKESTATION EDIT: Refactors Ventcawling to be less laggy
+		for(var/obj/machinery/atmospherics/member in range(8,get_turf(src)))
+			if(!istype(member))
+				continue
```

Comment:
without the `as anything` keyword, you will always receive the type of the loop iterator. This istype does nothing.

Reply:
Must've forgotten for the first part, but the second part is good to know